pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -49,9 +50,14 @@ func convertService(externalService *networking.ExternalService) []*model.Servic
 			svcPorts = append(svcPorts, convertPort(port))
 		}
 
+		// TODO(diemtvu) confirm this: ExternalName was not set before, but it was used
+		// (at least in v1) to decide if a service is external or not
+		// (https://github.com/istio/istio/blob/master/pilot/pkg/model/service.go#L417)
+		// Is this a bug for not setting it.
 		out = append(out, &model.Service{
 			MeshExternal: true,
 			Hostname:     host,
+			ExternalName: host,
```

Comment:
no need

Reply:
`ExternalName` is a legacy field that is only used in v1.